Packs/Cyberint/Integrations/Cyberint/Cyberint.py

```diff
@@ -538,167 +782,194 @@ def fetch_incidents(
-            alert['alert_data']['csv'] = extracted_csv_data
+            extracted_csv_data = extract_data_from_csv_stream(client, alert_id, alert_csv_id)  # type: ignore
+            alert["alert_data"]["csv"] = extracted_csv_data
 
         # add alert_name key to alert response
-        alert_name = f'Cyberint alert {alert_id}: {alert_title}'
-        alert.update({'alert_name': alert_name})
+        alert_name = f"Cyberint alert {alert_id}: {alert_title}"
+        alert.update({"alert_name": alert_name})
+
+        alert["closure_reason_description"] = "None"
+        alert["id"] = alert_id
+        alert["incident_id"] = alert_id
+        alert["mirror_direction"] = mirror_direction
+        alert["mirror_instance"] = demisto.integrationInstance()
 
         incident = {
-            'name': alert_name,
-            'occurred': datetime.strftime(alert_created_time, DATE_FORMAT),
-            'rawJSON': json.dumps(alert),
-            'severity': SEVERITIES.get(alert.get('severity', 'low')),
-            'attachment': incident_attachments,
+            "name": alert_name,
+            "occurred": datetime.strftime(alert_created_time, DATE_FORMAT),
+            "rawJSON": json.dumps(alert),
+            "severity": SEVERITIES.get(alert.get("severity", "low")),
+            "attachment": incident_attachments,
+            "mirror_direction": mirror_direction,
+            "mirror_instance": demisto.integrationInstance(),
         }
 
         if duplicate_alert and incident_csv_records:
             for index, incident_csv_record in enumerate(incident_csv_records):
-                alert_data.update({'content': incident_csv_record})
-                alert.update({'attachments': alert_data})
+                alert_data.update({"content": incident_csv_record})
+                alert.update({"attachments": alert_data})
 
-                alert_name = f'Cyberint alert {alert_id} ({index+1}): {alert_title}'
-                alert.update({'alert_name': alert_name})
+                alert_name = f"Cyberint alert {alert_id} ({index+1}): {alert_title}"
+                alert.update({"alert_name": alert_name})
 
-                incident.update({
-                    'name': alert_name,
-                    'rawJSON': json.dumps(alert)
-                })
+                incident.update({"name": alert_name, "rawJSON": json.dumps(alert)})
                 incidents.append(copy.deepcopy(incident))
         else:
             incidents.append(incident)
 
     if incidents:
         #  Update the time for the next fetch so that there won't be duplicates.
-        last_incident_time = incidents[0].get('occurred', '')
-        next_run = datetime.strptime(str(last_incident_time), DATE_FORMAT)
+        last_incident_time = incidents[0].get("occurred", "")
+        last_incident_time = max(incidents, key=lambda item: item["occurred"])
+        next_run = datetime.strptime(str(last_incident_time["occurred"]), DATE_FORMAT)
     next_run += timedelta(seconds=1)

```

Comment:
You have a point. This logic was not implemented by me but I will test this issue.

Reply:
Ok, I have been thinking about it. 
The timedelta is added to the last_run time to avoid duplicates. 
Cyberint don't support milliseconds in time arguments so I can only increase in one second.